index.test-d.ts

```diff
@@ -0,0 +1,9 @@
+import {expectType} from 'tsd-check';
+import pDefer, {DeferredPromise} from '.';
```

Comment:
Why does it work to import `DeferredPromise` even though it's not exported?

Reply:
You don't actually need to export interfaces fron an ambient declaration, they are exported automatically.

Reply:
Ugh, that is not very good behavior... It should be explicit.

Reply:
Yeah, forgot to export them...

Reply:
To be clear, I was talking about the ambient file behavior, not you. We should really have a lint rule for this.

Reply:
`dts-check` has one rule like this. Actually, `dts-check` has most rules that you'd need to cover your type definition styleguide.